clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -743,76 +743,76 @@ runSYCLPostLinkTool(ArrayRef<StringRef> InputFiles, const ArgList &Args) {
     if (!ImageFileOrErr)
       return ImageFileOrErr.takeError();
 
-    std::vector Modules = {module_split::SplitModule(
+    std::vector Modules = {module_split::ProcessedModule(
         *ImageFileOrErr, util::PropertySetRegistry(), "")};
     return Modules;
   }
 
-  return llvm::module_split::parseSplitModulesFromFile(*TempFileOrErr);
+  return llvm::module_split::parseProcessedModulesFromFile(*TempFileOrErr);
 }
 
-/// Invokes SYCL Split library for SYCL offloading.
+/// Invokes SYCL Post Link library for SYCL offloading.
 ///
 /// \param InputFiles the list of input LLVM IR files.
 /// \param Args Encompasses all arguments for linking and wrapping device code.
-///  It will be parsed to generate options required to be passed to SYCL split
-///  library.
+///  It will be parsed to generate options required to be passed to SYCL Post
+///  Link library.
 /// \param Mode The splitting mode.
 /// \returns The vector of split modules.
-static Expected<std::vector<module_split::SplitModule>>
-runSYCLSplitLibrary(ArrayRef<StringRef> InputFiles, const ArgList &Args,
-                    module_split::IRSplitMode Mode) {
-  std::vector<module_split::SplitModule> SplitModules;
-  if (DryRun) {
-    auto OutputFileOrErr = createOutputFile(
-        sys::path::filename(ExecutableName) + ".sycl.split.image", "bc");
-    if (!OutputFileOrErr)
-      return OutputFileOrErr.takeError();
-
-    StringRef OutputFilePath = *OutputFileOrErr;
-    auto InputFilesStr = llvm::join(InputFiles.begin(), InputFiles.end(), ",");
-    errs() << formatv("sycl-module-split: input: {0}, output: {1}\n",
-                      InputFilesStr, OutputFilePath);
-    SplitModules.emplace_back(OutputFilePath, util::PropertySetRegistry(), "");
-    return SplitModules;
-  }
-
-  llvm::module_split::ModuleSplitterSettings Settings;
+static Expected<std::vector<module_split::ProcessedModule>>
+runSYCLPostLinkLibrary(ArrayRef<StringRef> InputFiles, const ArgList &Args,
+                       module_split::IRSplitMode Mode) {
+  std::vector<module_split::ProcessedModule> OutputModules;
+  llvm::module_split::ModuleProcessingSettings Settings;
   Settings.Mode = Mode;
   Settings.OutputPrefix = "";
 
   for (StringRef InputFile : InputFiles) {
+    if (DryRun)
+      break;
```

Comment:
I find the previous version significantly more readable than the new one.

Reply:
Thank you for your thoughts.

Could you help me understand which aspects of the original code you find more readable? These 2 blocks (DryRun, Verbose) are almost identical and they take doubled attention.

Reply:
Okay, can we resolve this duplication by moving the duplicated code to a function? 
These two lines technically skip the whole loop, but it looks like we enter it. Now it's much more difficult to understand what happens when `DryRun` is `true`. The previous version has all code for that in a single scope statement. Now, I need to parse all the code of the function, which is written in unusual style.